runbotics/runbotics-scheduler/src/queue/queue.service.ts

```diff
@@ -91,15 +94,45 @@ export class QueueService implements OnModuleInit {
                 return Promise.reject(err);
             });
 
-        return job.finished() as Promise<StartProcessResponse>;
+            if (input?.timeout) {
```

Comment:
or maybe create a timeout before new queue job creation, pass timerId to it and remove the timer always when you finished this job